db/R/dbfiles.R

```diff
@@ -154,47 +162,53 @@ dbfile.input.insert <- function(in.path, in.prefix, siteid, startdate, enddate,
 ##' \dontrun{
 ##'   dbfile.input.check(siteid, startdate, enddate, 'application/x-RData', 'traits', dbcon)
 ##' }
-dbfile.input.check <- function(siteid, startdate=NULL, enddate=NULL, mimetype, formatname, parentid=NA, 
-                               con, hostname=fqdn(), ignore.dates=FALSE) {
-  if (hostname == "localhost") hostname <- fqdn();
-
-  mimetypeid <- get.id('mimetypes', 'type_string', mimetype, con = con)
+dbfile.input.check <- function(siteid, startdate = NULL, enddate = NULL, mimetype, 
+                               formatname, parentid = NA, con, hostname = fqdn(), ignore.dates = FALSE) {
+  if (hostname == "localhost") {
+    hostname <- fqdn()
+  }
+  
+  mimetypeid <- get.id("mimetypes", "type_string", mimetype, con = con)
   if (is.null(mimetypeid)) {
     return(invisible(data.frame()))
   }
   
   # find appropriate format
-  formatid <- get.id('formats', c("mimetype_id", "name"), c(mimetypeid, formatname), con) 
+  formatid <- get.id("formats", c("mimetype_id", "name"),
+                     c(mimetypeid, formatname), 
+                     con)
   if (is.null(formatid)) {
-    invisible(data.frame())
+    return(invisible(data.frame()))
   }
-
+  
   # setup parent part of query if specified
   if (is.na(parentid)) {
     parent <- ""
   } else {
     parent <- paste0(" AND parent_id=", parentid)
   }
-
+  
   # find appropriate input
-  if(ignore.dates) {
-    inputid <- db.query(paste0(
-      "SELECT id FROM inputs WHERE site_id=", siteid, " AND format_id=", formatid, parent), con)[['id']]
+  if (ignore.dates) {
+    inputid <- db.query(paste0("SELECT id FROM inputs WHERE site_id=", siteid, 
+                               " AND format_id=", formatid, parent), con)[["id"]]
   } else {
-    inputid <- db.query(paste0(
-      "SELECT id FROM inputs WHERE site_id=", siteid, " AND format_id=", formatid,
-      " AND start_date>='", startdate, "' AND end_date<='", enddate, "'", parent), con)[['id']]
+    inputid <- db.query(paste0("SELECT id FROM inputs WHERE site_id=", siteid, 
+                               " AND format_id=", formatid, 
+                               " AND start_date>='", startdate, 
+                               "' AND end_date<='", enddate, "'", parent), con)[["id"]]
   }
   if (is.null(inputid)) {
-    invisible(data.frame())
+    return(invisible(data.frame()))
```

Comment:
@robkooper agreed, it's not; but @mdietze and others have expressed preference for explicit `return`s, so I've started doing that.